engines/dfc_provider/app/services/affiliate_sales_data_builder.rb

```diff
@@ -0,0 +1,171 @@
+      sales_data
+        .uniq(&:producer_id)
+        .map { |sale| build_offer(sale) }
+    end
+
+    def build_order_lines
+      sales_data.map { |sale| build_order_line(sale) }
+    end
+
+    def build_orders
+      sales_data.map { |sale| build_order(sale) }
+    end
+
+    # def build_sale_sessions
+    #   sales_data.map { |sale| build_sale_session(sale) }
+    # end
+
+    private
+
+    def build_enterprise(id)
+      DataFoodConsortium::Connector::Enterprise.new(urls.enterprise_url(id))
+    end
+
+    def build_address(sale)
+      DataFoodConsortium::Connector::Address.new(
+        urls.address_url(sale.producer_id),
+        postalCode: sale.producer_postcode,
+        country: nil,
+        latitude: nil,
+        longitude: nil,
+        region: nil,
+        street: nil,
+        city: sale.producer_city
+      )
+    end
+
+    def build_producer(sale)
+      DataFoodConsortium::Connector::Enterprise.new(
+        urls.enterprise_url(sale.producer_id),
+        logo: nil,
+        name: nil,
+        description: nil,
+        vatNumber: nil,
+        suppliedProducts: build_supplied_product(sale),
+        localizations: build_address(sale)
+      )
+    end
+
+    def build_supplied_product(sale)
+      DataFoodConsortium::Connector::SuppliedProduct.new(
+        urls.enterprise_supplied_product_url(sale.producer_id, sale.product_id),
+        name: sale.product_name,
+        description: nil,
+        alcoholPercentage: nil,
+        lifetime: nil,
+        usageOrStorageConditions: nil,
+        totalTheoreticalStock: nil,
+        catalogItems: build_catalogue_item(sale),
+        quantity: build_quantity(sale)
+      )
+    end
+
+    def build_catalogue_item(sale)
+      DataFoodConsortium::Connector::CatalogItem.new(
+        urls.enterprise_catalog_item_url(sale.producer_id, sale.product_id),
+        stockLimitation: nil,
+        sku: nil,
+        offers: build_offer(sale)
+      )
+    end
+
+    def build_offer(sale)
+      DataFoodConsortium::Connector::Offer.new(
+        urls.enterprise_offer_url(sale.producer_id, sale.product_id),
+        stockLimitation: nil,
+      )
+    end
+
+    def build_order_line(sale)
+      DataFoodConsortium::Connector::OrderLine.new(
+        urls.enterprise_order_order_line_url(sale.producer_id, sale.order_id, sale.line_item_id),
+        description: nil,
+        offer: build_offer(sale),
+        order: build_order(sale),
+        quantity: build_quantity(sale),
+        price: build_price(sale)
+      )
+    end
+
+    def build_order(sale)
+      DataFoodConsortium::Connector::Order.new(
+        urls.enterprise_order_url(sale.producer_id, sale.order_id),
+        number: nil,
+        date: sale.order_date.strftime("%Y-%m-%d")
+        # saleSession: build_sale_session(sale)
+      )
+    end
+
+    def build_sale_session(sale)
+      DataFoodConsortium::Connector::SaleSession.new(
+        urls.enterprise_sale_session_url(sale.producer_id, sale.line_item_id),
```

Comment:
We were using it initially to add the distributor postcode but since the `SaleSession` has no support for `Coordination` I commented out the code that calls this method, for now. I think we will end up removing this code since we will most probably use the customer postcode as a proxy.